src/Stratis.Bitcoin/Configuration/NodeSettings.cs

```diff
@@ -222,71 +222,8 @@ public NodeSettings LoadArguments(string[] args)
             this.SyncTimeEnabled = config.GetOrDefault<bool>("synctime", true);
             this.Logger.LogDebug("Time synchronization with peers is {0}.", this.SyncTimeEnabled ? "enabled" : "disabled");
 
-            try
-            {
-                this.ConnectionManager.Connect.AddRange(config.GetAll("connect")
-                    .Select(c => ConvertIpAddressToEndpoint(c, this.Network.DefaultPort)));
-            }
-            catch (FormatException)
-            {
-                throw new ConfigurationException("Invalid 'connect' parameter.");
-            }
-
-            try
-            {
-                this.ConnectionManager.AddNode.AddRange(config.GetAll("addnode")
-                        .Select(c => ConvertIpAddressToEndpoint(c, this.Network.DefaultPort)));
-            }
-            catch (FormatException)
-            {
-                throw new ConfigurationException("Invalid 'addnode' parameter.");
-            }
-
-            var port = config.GetOrDefault<int>("port", this.Network.DefaultPort);
-            try
-            {
-                this.ConnectionManager.Listen.AddRange(config.GetAll("bind")
-                        .Select(c => new NodeServerEndpoint(ConvertIpAddressToEndpoint(c, port), false)));
-            }
-            catch (FormatException)
-            {
-                throw new ConfigurationException("Invalid 'bind' parameter");
-            }
-
-            try
-            {
-                this.ConnectionManager.Listen.AddRange(config.GetAll("whitebind")
-                        .Select(c => new NodeServerEndpoint(ConvertIpAddressToEndpoint(c, port), true)));
-            }
-            catch (FormatException)
-            {
-                throw new ConfigurationException("Invalid 'listen' parameter");
-            }
-
-            if (this.ConnectionManager.Listen.Count == 0)
-            {
-                this.ConnectionManager.Listen.Add(new NodeServerEndpoint(new IPEndPoint(IPAddress.Parse("0.0.0.0"), port), false));
-            }
-
-            var externalIp = config.GetOrDefault<string>("externalip", null);
-            if (externalIp != null)
-            {
-                try
-                {
-                    this.ConnectionManager.ExternalEndpoint = ConvertIpAddressToEndpoint(externalIp, port);
-                }
-                catch (FormatException)
-                {
-                    throw new ConfigurationException("Invalid 'externalip' parameter");
-                }
-            }
-
-            if (this.ConnectionManager.ExternalEndpoint == null)
-            {
-                this.ConnectionManager.ExternalEndpoint = new IPEndPoint(IPAddress.Loopback, this.Network.DefaultPort);
-            }
+            this.ConnectionManager = ConnectionManagerSettings.Load(this, this.ConnectionManager);
```

Comment:
@quantumagi it appears that we already create an instance of ConnectionManagerSettings in the constructor of NodeSettings. Therefore can we not just load the settings via that instance? Like so:

`this.ConnectionManager = this.ConnectionManager.Load(this);`

Then, in the Load method we don't need to check if ConnectionManagerSettings is null.

Reply:
I agree with @fassadlr why do it this way unless there is a reason to pass an existing instance.

Reply:
I am moving towards standardizing the parameters for the `Load` method across features. You have to look at DnsSettings to understand where this is headed. For the DnsSettings feature the purpose of the pattern would be to retain the value of the callback variable from the settings singleton and build on that object instead of creating a new settings object.

`            dnsSettings = dnsSettings ?? new DnsSettings();`